src/actions/write/setController.test.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import {
+  INVALID_INPUT_MESSAGE,
+  NON_CONTRACT_OWNER_MESSAGE,
+} from '../../../src/constants';
+import { AntContractWriteResult } from '../../../src/types';
+import { baselineAntState } from '../../../tests/utils/constants';
+import { setController } from './setController';
+
+describe('setController', () => {
+  let state = { ...baselineAntState };
```

Comment:
you can just spread `baselineAntState` in your tests rather than do this and the `beforeEach`